### DISCUSSION TITLE
How long are cached compilations stored?

### DISCUSSION THREAD

**Discussion by mbitsnbites** [compiler-explorer/compiler-explorer#3231] [Q&A]
I just noticed that the compilation results of a nightly/trunk toolchain reproduced results from yesterday (before the nightly update). When just changing a whitespace in the C++ source the newer version of the toolchain is used.

This leads me to believe that cached compilation results are used even for nightly/trunk versions?

Are cached results pruned after some time?

**Comment by partouf** [comment]
It's a bit complicated. If you hit a server instance where it was never seen before, the cache remains 2 days on our S3.
If you do hit an instance that's seen it before, it depends on how active that instance was and if it happened to have thrown it away.

**Comment by apmorton** [comment]
The S3 cache is always checked, so it will actually always stick around for
2 days currently. (But this is an implementation detail and subject to
change).

However, I thought our cache keys included the compiler binary mtime?

It seems only getCompilerCacheKey includes this, not getCacheKey.

Previously I think that would be sufficient, because rediscovery would run
the compiler and get a new fullVersion (which gets chucked into the
compiler object and _is_ used in the normal cache key).

I think with the compiler prediscovery this is now broken.

Some things to check:
- does the scheduled discovery still happen in prod?
- will it detect new compiler mtime?

If so, we should add mtime to the normal getCacheKey
